addons/website_slides_survey/tests/test_course_certification_failure.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.survey.tests.common import SurveyCase
+from odoo.addons.website_slides.tests.common import SlidesCase
+
+
+class TestCourseCertificationFailureFlow(SurveyCase):
```

Comment:
Well tests are a bit strange ... I understand you try to mix data coming from several separate test classes. However it seems to me create_slides_data is not really necessary. If some data is recreated here I think it is not really harmful it is just a slide and a channel ... those _data and _users methods make a lot of diff. For small tests it is better to keep things separated.

it seems surveycase is the most used class, let us inherit from that one and re-create channel data. Would that be complicated ?

One interesting thing would be to have a TestUserUtilityPhantomMixinToolsTestCase defined in base, and every major application could add a default test user to re-use. Maybe a task for master.

Reply:
I wanted so hard to reuse the "common" slides data that I made it look way more complicated that it needs to be....
I recreated the data as suggested to reduce the diff.